lib/travis/build/data.rb

```diff
@@ -144,7 +144,12 @@ def source_https?
       def source_ssh?
         return false if prefer_https?
         repo_private? && !installation? or
-          repo_private? && custom_ssh_key?
+          repo_private? && custom_ssh_key? or
+            force_private? && !installation?
```

Comment:
cool, here (L146 && L148) maybe (repo_private? || force_private?) && !installation? to prevent double call of installation?